### PR TITLE
fix(route-viewer): support fuzzy matching for non-numeric routes

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**

This PR adjusts the route search filtering to support fuzzy searching when the route doesn't have numbers. The library is the same one that's already included with otp-ui, so no bundle size increase.

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/util/state.js [hunk 1]
```diff
@@ -585,6 +586,24 @@ const getAgencyIdsForFilter = (name, transitOperators) => {
     .map((x) => x.agencyId)
 }
 
+/**
+ * Determine if a string matches a query. Uses direct matching if the string is only numbers,
+ * uses fuzzy matching otherwise.
+ */
+const isFilterMatch = (string, query) => {
+  const normalizedQuery = query.toLowerCase()
+  const normalizedString = string.toLowerCase()
```
Comment (amy-corson-ibigroup) [inline lib/util/state.js:595]: I'm wondering could we also filter dashes and spaces out of the search query + results? Just looking at the "1 line" search I know this PR inspired, and the results don't seem quite as intuitive as we'd hope. Removing spaces and punctuation could also allow for upping the similarity ratio from .4 to something a little higher? 
image

Comment (amy-corson-ibigroup) [reply lib/util/state.js:595]: Also so glad we talked about the correlation between hovering and these filters today, I thought I was going insane while reviewing this.

**Review by amy-corson-ibigroup — APPROVED** [review]
Great add! Looks solid to me, just a quick question 👍

File: lib/util/state.js (hunk 1 above)
Comment (amy-corson-ibigroup) [reply lib/util/state.js:595]: Sure sure, to be clear I think this works great as-is! But right now for this search, every result has a match of `.75`. If you remove spaces and dashes, `1-Line` has a match of `1`, but `T-Line` would still be `.75`

Comment (amy-corson-ibigroup) [reply lib/util/state.js:595]: Although I guess then you'd have to sort the results based on this percentage

Comment (miles-grant-ibigroup) [reply lib/util/state.js:595]: Yeah this issue is part of the reason why I was against doing fuzzy search way back when this was originally built. I think a text similar algorithm that could be tweaked for each set of routes might get the results that we're looking for, but yet another instance-specific method like that doesn't make me happy. I think our solution needs to be general

File: lib/util/state.js [hunk 2]
```diff
@@ -603,14 +622,9 @@ export const getFilteredRoutes = createSelector(
         (!filter.mode || filter.mode === route.mode) &&
         // If user search is active, filter by either the long or short name
```
Comment (binh-dam-ibigroup) [inline lib/util/state.js:623]: Update this comment to include any route description.

**Review by binh-dam-ibigroup — APPROVED** [review]
Cool stuff. Please add some examples in JSDoc.

File: lib/util/state.js [hunk 3]
```diff
@@ -603,14 +622,9 @@ export const getFilteredRoutes = createSelector(
...
         (!filter.search ||
-          (route.longName &&
-            route.longName
-              .toLowerCase()
-              .includes(filter.search.toLowerCase())) ||
-          (route.shortName &&
-            route.shortName
-              .toLowerCase()
-              .includes(filter.search.toLowerCase())))
+          (route.longName && isFilterMatch(route.longName, filter.search)) ||
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/util/state.js:625]: It shouldn't! OR is commutative is it not

**Pull request merged**